internal/v4/auth.go

```diff
@@ -141,17 +143,50 @@ func (h *Handler) authorizeWithPerms(req *http.Request, read, write []string, en
 	return err
 }
 
-const (
-	usernameAttr = "username"
-	groupsAttr   = "groups"
-)
+const usernameAttr = "username"
 
 // authorization conatains authorization information extracted from an HTTP request.
 // The zero value for a authorization contains no privileges.
 type authorization struct {
 	Admin    bool
 	Username string
-	Groups   []string
+}
+
+func (h *Handler) groupsForUser(username string) ([]string, error) {
+	if h.config.IdentityAPIURL == "" {
+		return nil, nil
+	}
+	// TODO cache groups for a user
+	url := h.config.IdentityAPIURL + "/v1/u/" + username + "/idpgroups"
```

Comment:
This can stay as /idpgroups until we introduce change in the IdM (agent gating) but then it will have to change into /groups endpoint.